### ISSUE TITLE
React$ElementConfig converts props to read-only

### ISSUE THREAD

**Issue by hrasoa** [facebook/flow#7220]
LazyPicure.js

```jsx
type Styles: {
  root: string,
};

type Props = {
  alt: string,
  image: { src: string, preSrc: string },
  styles: ?Styles,
};

const LazyPicture = (props: Props) => { ... }

LazyPicture.defaultProps = {
  styles: null,
};
```

index.js

```jsx
import Loadable from 'react-loadable';
import typeof LazyPicture from './LazyPicture';

type Props = React$ElementConfig<LazyPicture>;

const LoadableLazyPicture: React$ComponentType<Props> = Loadable({
  loader: () => import(/* webpackChunkName: "lazy-picture" */ './LazyPicture'),
  loading: () => null,
});

export default LoadableLazyPicture;
```

Now when I run flow:

<img width="853" alt="capture d ecran 2018-11-30 a 14 15 13" src="https://user-images.githubusercontent.com/796524/49291392-7804d380-f4aa-11e8-8e08-cb34282011a5.png">

In fact in index.js, ```type Props``` from ```React$ElementConfig``` became read only:
```jsx
type Props = {
    +alt: string,
    +image: {
        preSrc: string,
        src: string
    },
    +styles ? : ? Styles
}
```

**I expected to have the same type from ```React$ElementConfig```**

I cannot use ```React$ElementProps``` because I need to preserve the ```defaultProps```

```
flow-bin@0.87.0
```

https://flow.org/en/docs/react/types/#toc-react-elementprops
https://flow.org/en/docs/react/types/#toc-react-elementconfig

**Comment by jbrown215** [comment]
Would you mind also showing the signature for loadable? This read-only conversion is by design. It's also likely that this issue will be fixed by something I am landing in 0.89.0, so I'd like to confirm with your example.

**Comment by hrasoa** [comment]
@jbrown215 here: https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/react-loadable_v5.x.x/flow_v0.56.0-/react-loadable_v5.x.x.js

**Comment by jbrown215** [comment]
[Small repro](https://flow.org/try/#0C4TwDgpgBAwgCgJwPZgM5QLxQN4CgpSqgA2EqAXFAPxEICWAdgOYA0uAvgNy64DGSDIlAAyAQwBeIOHV7AArgmhYAFGGRpK8dagCUmAHw4oi+QgZQGc4sSjseYydNkKIAOgAmEAGairwRCjoWHgERCCkFBZWxGxcPJ68xKKKUABuySJIou6iAEaklAA8ACoBaPrKehiGMEmoqIXCWTn5EDBIALZgAhAMwCVlqPr63LigkFCDmFAAShCisgAkAKKkHb3A7QxedEyF4xBIXiISUjKmECP4BFC4CUkpiaL1mdl5pO1dPX0D2oYQAA9gL13Og5gtgItPt0GBtfoFDNh2Ddbii+AIhE03q0HGdnIpKOCltDvsBiuAIIVBoYsFiWqRCloEZVOEA).

I can confirm that this error goes away with the changes that will be released in 0.89.0.

**Comment by hrasoa** [comment]
@jbrown215 good to know, thanks
